test/IISIntegration.FunctionalTests/Utilities/SkipIfNotAllowedToRunIISAttribute.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Security.Principal;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Win32;
+
+namespace Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests
+{
+    [AttributeUsage(AttributeTargets.Assembly | AttributeTargets.Class | AttributeTargets.Method)]
+    public sealed class SkipIfNotAllowedToRunIISAttribute : Attribute, ITestCondition
+    {
+        public bool IsMet {
+            get
+            {
+                // Two conditions need to be met:
+                // IIS needs to be installed
+                var identity = WindowsIdentity.GetCurrent();
+                var principal = new WindowsPrincipal(identity);
+                var isInrole = principal.IsInRole(WindowsBuiltInRole.Administrator);
+
+                var w3wpExists = File.Exists(Path.Combine(Environment.SystemDirectory, "Windows", "System32", "inetsrv", "w3wp.exe"));
+                return w3wpExists && isInrole;
+            }
+        }
+
+        public string SkipReason { get; } = $"Hostable Web Core not availible, {TestServer.HostableWebCoreLocation} not found.";
```

Comment:
Change this to be no w3wp exist or not admin depending on why we are skipping

Reply:
copy pasta.